modelstore/storage/azure.py

```diff
@@ -143,10 +147,16 @@ def _get_storage_location(self, meta: dict) -> str:
         return meta["prefix"]
 
     def _read_json_objects(self, path: str) -> list:
+        logger.debug("Listing files in: %s/%s", self.container_name, path)
         results = []
         blobs = self._container_client().list_blobs(name_starts_with=path + "/")
         for blob in blobs:
             if not blob.name.endswith(".json"):
+                logger.debug("Skipping non-json file: %s", blob.name)
+                continue
+            if os.path.split(blob.name)[0] != path:
+                # We don't want to read files in a sub-prefix
+                logger.debug("Skipping file in sub-prefix: %s", blob.name)
```

Comment:
Same problem in Azure as seen in AWS, above. This did not affect Google Cloud or local file systems though.